src/azul/chalice.py

```diff
@@ -182,11 +179,22 @@ def _register_handler(self,
                                   wrapped_handler, kwargs, options)
         # Our handlers reference the name of the corresponding Lambda function
         # which allows the handler to be the single source of truth when
-        # configuring Terraform, etc.
-        if hasattr(wrapped_handler, 'lambda_name'):
-            assert wrapped_handler.lambda_name == name
-        else:
-            wrapped_handler.lambda_name = name
+        # configuring Terraform, etc. We store other parameters used to
+        # configure the handler for the same reason.
+        attributes = ('queue',)
+        for attribute, value in (('lambda_name', name),
+                                 *((key, kwargs[key]) for key in attributes if key in kwargs)):
+            if hasattr(wrapped_handler, attribute):
+                # Attributes may already exist when a handler function has
+                # multiple Chalice decorators (usually route decorators).
+                assert getattr(wrapped_handler, attribute) == value
+            else:
+                setattr(wrapped_handler, attribute, value)
+        if 'path' in kwargs:
+            if hasattr(wrapped_handler, 'path'):
+                wrapped_handler.paths.append(kwargs['path'])
+            else:
+                wrapped_handler.paths = [kwargs['path']]
```

Comment:
This is an odd mix of general and specific code and it doesn't use EAFP. Can you try this:

```python
Index: src/azul/chalice.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/chalice.py b/src/azul/chalice.py
--- a/src/azul/chalice.py	(revision d327e6a6aeeb0dbe6930695cd24c9d0c8410dd54)
+++ b/src/azul/chalice.py	(date 1618509062574)
@@ -168,6 +168,8 @@
         else:
             log.info('Returning %i response. To log headers and body, set AZUL_DEBUG to 1.', response.status_code)
 
+    absent = object()
+
     def _register_handler(self,
                           handler_type,
                           name,
@@ -181,20 +183,23 @@
         # which allows the handler to be the single source of truth when
         # configuring Terraform, etc. We store other parameters used to
         # configure the handler for the same reason.
-        attributes = ('queue',)
-        for attribute, value in (('lambda_name', name),
-                                 *((key, kwargs[key]) for key in attributes if key in kwargs)):
-            if hasattr(wrapped_handler, attribute):
-                # Attributes may already exist when a handler function has
-                # multiple Chalice decorators (usually route decorators).
-                assert getattr(wrapped_handler, attribute) == value
-            else:
-                setattr(wrapped_handler, attribute, value)
-        if 'path' in kwargs:
-            if hasattr(wrapped_handler, 'path'):
-                wrapped_handler.paths.append(kwargs['path'])
-            else:
-                wrapped_handler.paths = [kwargs['path']]
+        for attribute, new_value, is_additive in [
+            ('queue', kwargs.get('queue', self.absent), False),
+            ('lambda_name', name, False),
+            ('path', kwargs.get('path', self.absent), True)
+        ]:
+            if new_value is not self.absent:
+                try:
+                    old_value = getattr(wrapped_handler, attribute)
+                except AttributeError:
+                    if is_additive:
+                        new_value = [new_value]
+                    setattr(wrapped_handler, attribute, new_value)
+                else:
+                    if is_additive:
+                        old_value.append(new_value)
+                    else:
+                        assert old_value == new_value
 
     # Some type annotations to help with auto-complete
     lambda_context: LambdaContext
```

If you think we can conflate None and absence, you can drop the `self.absent` stuff.

Reply:
I like this way better.

> If you think we can conflate None and absence, you can drop the self.absent stuff.

Probably conflating is OK, but better safe than sorry. I like this pattern anyway.